libs/platform/src/lib/table-helpers/directives/table-virtual-scroll.directive.ts

```diff
@@ -79,62 +135,174 @@ export class TableVirtualScrollDirective extends TableVirtualScroll implements O
+            this._table.setRowsInViewport(
+                startNodeIndex,
+                rowsVisible.slice(startNodeIndex, startNodeIndex + visibleNodeCount).length
+            );
+        }
     }
 
     /**
      * Initialises scroll listener.
      */
     listenOnVirtualScroll(): void {
-        this._tableScrollDispatcher
-            .verticallyScrolled()
-            .pipe(
-                filter(() => this.virtualScroll && !!this.bodyHeight),
-                takeUntil(this._destroy$)
-            )
-            .subscribe((scrollable) => {
-                this.calculateVirtualScrollRows();
-                this._table.setTableState({
-                    ...this._table.getTableState(),
-                    scrollTopPosition: scrollable.getScrollTop()
-                });
+        if (this.scrollWholeRows) {
+            this._table.tableContainer.nativeElement.addEventListener('wheel', this._wheelScrollListenerFunction, {
+                passive: false
             });
+            this._table.tableScrollMockContainer.nativeElement.addEventListener(
+                'scroll',
+                this._mockScrollbarListenerFunction,
+                { passive: false }
+            );
+        } else {
+            this._tableScrollDispatcher
+                .verticallyScrolled()
+                .pipe(
+                    filter(() => this.virtualScroll && !!this.bodyHeight),
+                    takeUntil(this._destroy$)
+                )
+                .subscribe((scrollable) => {
+                    this.calculateVirtualScrollRows();
+                    this._table.setTableState({
+                        ...this._table.getTableState(),
+                        scrollTopPosition: scrollable.getScrollTop()
+                    });
+                });
+        }
+    }
+
+    /** @hidden */
+    private _getNumberOfRowsToDisplay(): number {
+        let tableHeight = this._table.tableContainer.nativeElement.clientHeight;
+        tableHeight = tableHeight - this._table.tableContainer.nativeElement.querySelector('thead').clientHeight;
+        return Math.floor(tableHeight / (this.rowHeight + 2));
+    }
+
+    /** @hidden */
+    private _wheelScrollListenerFunction = (event: WheelEvent): void => {
+        if (Math.abs(event.deltaX) < Math.abs(event.deltaY)) {
+            event.preventDefault();
+            event.stopImmediatePropagation();
+            clearTimeout(this._wheelTimeout);
+            const deltaY = event.deltaY;
+            if (deltaY) {
+                this._wheelTimeout = setTimeout(() => {
+                    this._table.tableScrollMockContainer.nativeElement.scrollBy({ top: deltaY });
+                    this._scrollRow(deltaY > 0 ? 'down' : 'up', deltaY > 0 ? 1 : -1);
+                    this._lastMockScrollPosition = this._table.tableScrollMockContainer.nativeElement.scrollTop;
+                }, 5);
```

Comment:
duplicated DOM query. Also is it better to use https://developer.mozilla.org/en-US/docs/Web/API/Window/requestAnimationFrame instead of setTimeout?